cborg/src/Codec/CBOR/Decoding.hs

```diff
@@ -423,7 +424,7 @@ decodeWord64 =
 #if defined(ARCH_64bit)
   Decoder (\k -> return (ConsumeWord (\w# -> k (toWord64 w#))))
 #else
-  Decoder (\k -> return (ConsumeWord64 (\w64# -> k (toWord64 w64#))))
+  Decoder (\k -> return (ConsumeWord64 (\w64# -> k (W64# w64#))))
```

Comment:
I don't understand why we're replacing our conversion helper function `toWord64` with a concrete implementation. And indeed deleting the helper function. Why not just make the helper function do the right thing?

Reply:
Will change, see https://github.com/well-typed/cborg/pull/322/files#r1320790119